### PR TITLE
Refactor client and add in publisher to Mr.Robot

### PR DESCRIPTION
**Pull request by herman-rogers**

### CODE REVIEW

File: client/gogeta_client.go [hunk 1]
```diff
@@ -41,39 +55,121 @@ func (client *GogetaClient) Start() {
...
+func (client *Gogeta) archiveRepo(repo *sourcesystem.SourceRepository) {
+	archive := path.Join(os.Getenv("GOPATH"), "/repos/", repo.ProjectName+".zip")
+	storageData := storehouse.StorageData{
+		Source: repo.SourceLocation,
+		Target: archive,
+	}
+	if err := client.Storage.StoreFiles(&storageData); err != nil {
+		client.Log.Error(err.Error())
+	}
+	client.Data[0].ArchivePath = repo.ProjectName + ".zip"
```
Comment (herman-rogers) [inline client/gogeta_client.go]: I'll add this to a variable, just noticed

File: client/gogeta_client.go [hunk 2]
```diff
@@ -41,39 +55,121 @@ func (client *GogetaClient) Start() {
...
+		BuildID:        data.BuildID,
+	}
+	jsonMessage, err := json.Marshal(message)
+	if err != nil {
+		client.Log.Error(err.Error())
+		return
+	}
+	notification := publisher.Message{
+		JSON:     jsonMessage,
+		Subject:  "Buildr Request",
```
Comment (herman-rogers) [inline client/gogeta_client.go:176]: What should I call this topic for Mr. Robot?

File: pkg/publisher/simple_notification.go [hunk 3]
```diff
@@ -1,18 +1,24 @@
...
 
 // SendJSON sends a strigified json object as Message to an endpoint
-func (service SimpleNotification) SendJSON(msg *Message) {
+func (service *SimpleNotification) SendJSON(msg *Message) {
 	msg.Message = string(msg.JSON)
 	_, err := service.Application.PublishMessage(*msg)
 	if err != nil {
+		if service.Log == nil {
+			fmt.Printf(err.Error())
+			return
```
Comment (herman-rogers) [inline pkg/publisher/simple_notification.go:20]: print out any errors on the off chance logger is nil.

File: client/gogeta_client.go [hunk 4]
```diff
@@ -41,39 +55,121 @@ func (client *GogetaClient) Start() {
...
+	// queue system
+	sess := session.Must(session.NewSession())
+	amazonQueue := queuesystem.AmazonQueue{
 		Client: sqs.New(sess),
-		Region: os.Getenv(config.Region),
 		URL:    os.Getenv(config.QueueURL),
 	}
+
+	// publisher system
+	amazonSNS := publisher.AmazonNotification{}
```
Comment (chvck) [inline client/gogeta_client.go:70]: should the client be apathetic about what the notification system is?

File: client/gogeta_client.go [hunk 5]
```diff
@@ -41,39 +55,121 @@ func (client *GogetaClient) Start() {
...
 		client.Log.Error(err.Error())
 	}
+	client.Data = messages
+}
 
-	if len(messages) <= 0 {
-		return &repo
+func (client *Gogeta) setVersionControl() {
+	dataType := client.Data[0].Type
+	switch dataType {
```
Comment (chvck) [inline client/gogeta_client.go]: should we do a toUpper or toLower (or whatever in go) on this?

File: client/gogeta_client.go (hunk 2 above)
Comment (chvck) [reply client/gogeta_client.go:176]: Don't think it matters does it? I'm not sure I ever actually use the message topic.

File: client/gogeta_client.go (hunk 5 above)
Comment (herman-rogers) [reply client/gogeta_client.go]: sure we can force it to upper or w/e

File: client/gogeta_client.go (hunk 4 above)
Comment (herman-rogers) [reply client/gogeta_client.go:70]: it essentially is, I haven't made an abstract client thing yet, I've just glued the pieces on top. For now we'll only have the gogeta client but in the future I can make this more general.

File: client/gogeta_client.go (hunk 2 above)
Comment (herman-rogers) [reply client/gogeta_client.go:176]: I don't think so either

File: client/gogeta_client.go [hunk 6]
```diff
@@ -13,26 +17,40 @@ import (
...
 
-// GogetaClient is the high level implementation of the gogeta
-// source control system
-type GogetaClient struct {
-	Queue   queuesystem.Messages
-	Log     logger.Log
-	SCM     sourcesystem.SourceSystem
-	Storage storehouse.StoreHouse
+// Gogeta is the source control manager implementation
+type Gogeta struct {
```
Comment (chvck) [inline client/gogeta_client.go:21]: Do all of these properties need to be exposed? If they're just for testing then we could just use internal tests on it?

Comment (herman-rogers) [reply client/gogeta_client.go:21]: Data doesn't need to be exposed, but the systems need to be exposed so we can mock out / reuse the GogetaClient (specify outside of this scope)

**Pull request merged**